src/main/resources/assets/carpet/scripts/distance_rc_1.sc

```diff
@@ -0,0 +1,88 @@
+import('math','_euclidean','_manhattan','_round');
+
+__config()->{
+    'commands'->{
+        'from <start_pos>'->'set_start',
+        'from <start_pos> to <end_pos>'->'_calculate',
+        'to <end_pos>'->_(end_pos)->(_calculate(null,end_pos)),
+        'toggle_line'->'_toggle_line'
+    },
+    'scope'->'player'
+};
+
+global_position=null;
+global_end_position=null;
+
+global_show_line=false;
+
+global_line = {
+    'start_pos'->global_position,
+    'end_pos'->global_end_position
+};
+
+set_start(pos)->(
+    global_position= pos;
+    print(player(),format('gi Initial point set to: ', 'g '+pos,'?/tp '+(str(pos)-','-'['-']')))
+);
+
+_toggle_line()->(
+    global_show_line= !global_show_line;
+    if(global_show_line,
+        print('Toggled lines on');
+        if(global_position && global_end_position,_draw_line()),//making sure that neither of them are null
```

Comment:
This would cause the message to be sent without the line being actually scheduled for next time. Also the variable would have changed too, so next time you run the command it would first say "turned off" and you would need to run it again.

I'd suggest to check those either when drawing (more flexible) or before even toggling the variable.

Also you only need to check for `end_position` since you never set it without start.

Reply:
nah, it works good. Ill take that last suggestion tho, tru.